### PR TITLE
Feat/core types 6

### PR DESCRIPTION
**Pull request by justinkaseman**
## Description
Updates from requested changes & merging develop

### CODE REVIEW

File: packages/core/bootstrap/test/unit/cache-warmer.test.ts [hunk 1]
```diff
@@ -26,11 +34,9 @@ beforeEach(() => {
...
 })
 
-function stateStream(initialWarmerState: {
-  cacheWarmer: CacheWarmerState
-}): StateObservable<RootState> {
-  return new StateObservable<RootState>(new Subject(), {
-    ...initialState,
+function stateStream(initialWarmerState: { cacheWarmer: CacheWarmerState }): StateObservable<any> {
+  return new StateObservable<any>(new Subject(), {
+    // ...initialState,
```
Comment (boxhock) [inline packages/core/bootstrap/test/unit/cache-warmer.test.ts:39]: Is this supposed to be commented out?

File: packages/sources/renvm-address-set/test/integration/adapter.test.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ describe('execute', () => {
         .set('Accept', '*/*')
         .set('Content-Type', 'application/json')
         .expect('Content-Type', /json/)
-        .expect(200)
+      // .expect(200)
```
Comment (boxhock) [inline packages/sources/renvm-address-set/test/integration/adapter.test.ts:52]: Intentional?

**Review by boxhock — COMMENTED** [review]
Some comments, but other than that LGTM

File: packages/composites/bsol-price/test/integration/price.test.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ describe('accounts', () => {
         .set('Accept', '*/*')
         .set('Content-Type', 'application/json')
         .expect('Content-Type', /json/)
-        .expect(200)
+      // .expect(200)
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply packages/composites/bsol-price/test/integration/price.test.ts:63]: Nope

File: packages/core/bootstrap/test/unit/cache-warmer.test.ts (hunk 1 above)
Comment (justinkaseman) [reply packages/core/bootstrap/test/unit/cache-warmer.test.ts:39]: No, good catch

File: packages/sources/renvm-address-set/test/integration/adapter.test.ts (hunk 2 above)
Comment (justinkaseman) [reply packages/sources/renvm-address-set/test/integration/adapter.test.ts:52]: I do have a gripe with these though. They stifle the error. We should take all of them off the nock expect that put them in their own jest assertion. For another PR though.

File: packages/composites/apy-finance/src/endpoint/allocations.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const inputParameters: InputParameters<TInputParameters> = {}
 const getAllocations = async (registry: ethers.Contract): Promise<types.TokenAllocations> => {
   const allocationIds = await registry.getAssetAllocationIds()
   const [components, balances, decimals] = await Promise.all([
-    Promise.all<string[]>(allocationIds.map((id: string) => registry.symbolOf(id))),
+    Promise.all<string>(allocationIds.map((id: string) => registry.symbolOf(id))),
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply packages/composites/apy-finance/src/endpoint/allocations.ts:18]: It's actually reversed, the other ones shouldn't have `[]`.

So promise all returns an array of promises - you can imagine the map returning an array of `[registry.symbolOf(id), registry.symbolOf(id), ...]`. That type being passed in is the return type of _one_ of the promises. It adds the array for you. When it settles it turns into an array of strings.

[the remark this answers is not in the document]
Comment (boxhock) [reply packages/composites/apy-finance/src/endpoint/allocations.ts:18]: Ah I see, will you remove the `[]` on the other ones in this PR as well, or are we keeping them like this?

[the remark this answers is not in the document]
Comment (justinkaseman) [reply packages/composites/apy-finance/src/endpoint/allocations.ts:18]: Yes, they are removed in https://github.com/smartcontractkit/external-adapters-js/pull/1952/commits/0b2e458d4ec8aa576b70ea8e400235bd22d4ce34

[the remark this answers is not in the document]
Comment (boxhock) [reply packages/composites/apy-finance/src/endpoint/allocations.ts:18]: Got it, ty!

**Pull request merged**